wc3-hw1-TranNgocLam/cypress/e2e/registrationForm.cy.js

```diff
@@ -0,0 +1,308 @@
+
+  // TCRegister#2: Verify that user can submit successfully with valid data in required data fields
+  it('TCRegister#2', () => {
+    cy.get('@registration').then((users) => {
+
+      registerForm
+        .inputLastName(users[1].lastName)
+        .inputFirstName(users[1].firstName)
+        .checkGender(users[1].gender)
+        .inputUserNumber(users[1].userNumber)
+        .clickSubmit()
+
+      confirmForm
+        .isCorrectName(users[1].firstName, users[1].lastName)
+        .isCorrectGender(users[1].gender)
+        .isCorrectMobile(users[1].userNumber)
+        .clickClose()
+    });
+  });
+
+  /*   TCRegister#3: Verify that user submit successfully when first name, last name, current adrress have abundant space  at the begin and end of input data:
+     - First name, Last name, Current Adress have more than 255 characters.
+     - Picture file's size exceeds limit file size 2MB. */
+  it('TCRegister#3', () => {
+    cy.get('@registration').then((users) => {
+
+      registerForm
+        .inputLastName(users[0].lastName)
+        .inputFirstName(users[0].firstName)
+        .checkGender(users[0].gender)
+        .inputUserNumber(users[0].userNumber)
+        .clickSubmit()
+
+      confirmForm
+        .isCorrectName(users[0].firstName, users[0].lastName)
+        .isCorrectGender(users[0].gender)
+        .isCorrectMobile(users[0].userNumber)
+        .clickClose()
+    });
+  });
+
+  // TCRegister#4: Verify that user cannot submit with empty data fields.
+  it('TCRegister#4', () => {
+    registerForm
+      .clickSubmit()
+      .checkFirstNameValid(false)
+      .checkLastNameValid(false)
+      .isGenderChecked(false)
+      .checkMobileValid(false)
+  });
+
+    /*  TCRegister#5: Verify that user cannot submit with invalid data:
+      - First Name includes numbers and special characters.
+      - Last Name includes numbers and special characters */
+    it.only('TCRegister#5', () => {
+      cy.get('@registration').then((users) => {
+        registerForm
+          .inputLastName(users[2].lastName)
+          .inputFirstName(users[2].firstName)
+          .checkGender(users[2].gender)
+          .inputUserNumber(users[2].userNumber)
+          .clickSubmit()
+
+        confirmForm
+          .isCorrectName(users[2].firstName, users[2].lastName)
+          .isCorrectGender(users[2].gender)
+          .isCorrectMobile(users[2].userNumber)
+          .clickClose()
+      });
+    });
+
+    /*  TCRegister#6: Verify that user cannot submit with invalid data:
+      - Email lacks of '@' and email domain.
+      - Phone number has less than 10 digits. */
+    it.only('TCRegister#6', () => {
+      cy.get('@registration').then((users) => {
+        registerForm
+          .inputLastName(users[3].lastName)
+          .inputFirstName(users[3].firstName)
+          .inputEmail(users[3].email)
+          .checkGender(users[3].gender)
+          .inputUserNumber(users[3].userNumber)
+          .clickSubmit()
+
+        registerForm
```

Comment:
You can chain the action steps with the verify steps if they are in a same page, it's ok